### DISCUSSION TITLE
Updating SASS

### DISCUSSION THREAD

**Discussion by orangecoloured** [sveltejs/kit#12710] [Q&A] [answered]
After the recent update of the `sass` package I started to get these warning while building my app.
`Deprecation Warning: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.`

I updated `vite.config.ts` like this, but still getting the warning.
```
css: {
  preprocessorOptions: {
    sass: {
      api: 'modern'
    }
  }
}
```

How do I fix this?

**Comment by mojoaxel** [comment, marked as answer]
For me this stack-overflow answer solved this problem:

> The default API when using Vite is currently set to legacy, but you can change it in your vite.config.[js/ts].
>
> ```js
> export default defineConfig({
>  css: {
>    preprocessorOptions: {
>      scss: {
>        api: 'modern-compiler' // or "modern"
>      }
>    }
>  }
>})
>
>Reference: https://vitejs.dev/config/shared-options#css-preprocessoroptions

https://stackoverflow.com/a/79003101/722162 ([CC-BY-SA](https://creativecommons.org/licenses/by-sa/4.0/) by [hiws](https://stackoverflow.com/users/8397974/hiws))
Tested with `svelte@4.2.19` and `vite@5.4.8`.

**Reply by orangecoloured** [reply]
Oh, wow. For some reason the first time I tried it, it didn't work. Now it worked on the first try.
Thanks!

**Reply by cheekybuddha** [reply]
Property:
sass vs scss
🤓😂

**Comment by farhansadikgalib** [comment]
Hi brothers, I need some stars for my open source contribution. I would appreciate it if everyone could add a star to this repository.
https://github.com/farhansadikgalib/dropdown_flutter

**Reply by mojoaxel** [reply]
@farhansadikgalib Please stop spamming other projects. This is not appreciated!
